extractor/src/main/java/org/schabi/newpipe/extractor/utils/HttpUtils.java

```diff
@@ -0,0 +1,36 @@
+package org.schabi.newpipe.extractor.utils;
+
+import java.util.Arrays;
+
+import org.schabi.newpipe.extractor.downloader.Response;
+import org.schabi.newpipe.extractor.exceptions.HttpResponseException;
+
+public final class HttpUtils {
+
+    private HttpUtils() {
+        // Utility class, no instances allowed
+    }
+    
+    // CHECKSTYLE:OFF
+    /**
+     * Validates the response codes for the given {@link Response}, and throws 
+     * a {@link HttpResponseException} if the code is invalid
+     * @param response The response to validate
+     * @param validResponseCodes Expected valid response codes
+     * @throws HttpResponseException Thrown when the response code is not in {@code validResponseCodes},
+     * or when {@code  validResponseCodes} is empty and the code is a 4xx or 5xx error.
+     */
+    // CHECKSTYLE:ON
+    public static void validateResponseCode(final Response response,

```

Comment:
I would inline this directly in `Response`, especially because it's first argument is a `Response`